pype/plugins/global/publish/integrate_new.py

```diff
@@ -91,18 +94,28 @@ class IntegrateAssetNew(pyblish.api.InstancePlugin):
     default_template_name = "publish"
     template_name_profiles = None
 
-    def process(self, instance):
+    # file_url : file_size of all published and uploaded files
+    integrated_file_sizes = {}
+
+    TMP_FILE_EXT = 'tmp'  # suffix to denote temporary files, use without '.'
 
+    def process(self, instance):
+        self.integrated_file_sizes = {}
         if [ef for ef in self.exclude_families
                 if instance.data["family"] in ef]:
             return
 
-        self.register(instance)
-
-        self.log.info("Integrating Asset in to the database ...")
-        self.log.info("instance.data: {}".format(instance.data))
-        if instance.data.get('transfer', True):
-            self.integrate(instance)
+        try:
+            self.register(instance)
+            self.log.info("Integrated Asset in to the database ...")
+            self.log.info("instance.data: {}".format(instance.data))
+            self.handle_destination_files(self.integrated_file_sizes,
+                                          'finalize')
+        except Exception as e:
+            # clean destination
+            self.log.critical("Error when registering",  exc_info=True)
+            self.handle_destination_files(self.integrated_file_sizes, 'remove')
+            raise
```

Comment:
We need to maintain compatiblity with python 2.7 but in this case I think re-raise feature is usefull to maintain track. We have **six** compatibility library in dependecies so I would use [six.reraise()](https://six.readthedocs.io/#six.reraise).

Reply:
Ok, changed all existing too for conformity